packages/material-ui-lab/src/YearPicker/PickersYear.tsx

```diff
@@ -15,90 +26,132 @@ export interface YearProps {
-    width: 72,
-    borderRadius: 16,
-    cursor: 'pointer',
+  [`&.${pickersYearClasses.selected}`]: {
+    color: theme.palette.primary.contrastText,
+    backgroundColor: theme.palette.primary.main,
     '&:focus, &:hover': {
-      backgroundColor: alpha(theme.palette.action.active, theme.palette.action.hoverOpacity),
-    },
-    '&$disabled': {
-      color: theme.palette.text.secondary,
-    },
-    '&$selected': {
-      color: theme.palette.primary.contrastText,
-      backgroundColor: theme.palette.primary.main,
-      '&:focus, &:hover': {
-        backgroundColor: theme.palette.primary.dark,
-      },
+      backgroundColor: theme.palette.primary.dark,
     },
   },
-  disabled: {},
-  selected: {},
-});
+}));
 
 /**
  * @ignore - internal component.
  */
-const PickersYear = React.forwardRef<HTMLButtonElement, YearProps & WithStyles<typeof styles>>(
-  (props, forwardedRef) => {
-    const { autoFocus, classes, children, disabled, onClick, onKeyDown, selected, value } = props;
-    const ref = React.useRef<HTMLButtonElement>(null);
-    const refHandle = useForkRef(ref, forwardedRef as React.Ref<HTMLButtonElement>);
-    const wrapperVariant = React.useContext(WrapperVariantContext);
+const PickersYear = React.forwardRef<HTMLButtonElement, YearProps>(function PickersYear(
+  inProps,
+  forwardedRef,
+) {
+  const props = useThemeProps({ props: inProps, name: 'MuiPickersYear' });
+  const { autoFocus, className, children, disabled, onClick, onKeyDown, selected, value } = props;
+  const ref = React.useRef<HTMLButtonElement>(null);
+  const refHandle = useForkRef(ref, forwardedRef as React.Ref<HTMLButtonElement>);
+  const wrapperVariant = React.useContext(WrapperVariantContext);
+
+  const styleProps = {
+    ...props,
+    wrapperVariant,
+  };
 
-    // TODO: Can we just forward this to the button?
-    React.useEffect(() => {
-      if (autoFocus) {
-        // `ref.current` being `null` would be a bug in Material-UIu
-        ref.current!.focus();
-      }
-    }, [autoFocus]);
+  const classes = useUtilityClasses(styleProps);
 
-    return (
-      <div
-        data-mui-test="year"
-        className={clsx(classes.root, {
-          [classes.modeDesktop]: wrapperVariant === 'desktop',
-        })}
+  // TODO: Can we just forward this to the button?
+  React.useEffect(() => {
+    if (autoFocus) {
+      // `ref.current` being `null` would be a bug in Material-UIu
+      ref.current!.focus();
+    }
+  }, [autoFocus]);
+
+  return (
+    <PickersYearRoot
+      data-mui-test="year"
+      className={clsx(classes.root, className)}
+      styleProps={styleProps}
+    >
+      <PickersYearButton
+        ref={refHandle}
+        disabled={disabled}
+        type="button"
+        data-mui-test={`year-${children}`}
+        tabIndex={selected ? 0 : -1}
+        onClick={(event) => onClick(event, value)}
+        onKeyDown={(event) => onKeyDown(event, value)}
+        className={classes.yearButton}
+        styleProps={styleProps}
```

Comment:
@mnajdova How could I make sure that this prop is required? Not adding it would be a mistake here. Making it required at the type level would guard against mistakes and we can drop optional chaining in https://github.com/mui-org/material-ui/pull/25949/files#diff-a4a95615f7e88d08780b14afe799a9cfa941f40d20b9cc7bf5fae22e95943fc5R67

Reply:
I believe we can make it required everywhere. We should update https://github.com/mui-org/material-ui/blob/next/packages/material-ui/src/styles/experimentalStyled.d.ts

Reply:
It may be awkward for the developers using the `styled()` util, as it would be required everywhere tough..

Reply:
Or, how about we do
```diff --git a/packages/material-ui-lab/src/YearPicker/PickersYear.tsx b/packages/material-ui-lab/src/YearPicker/PickersYear.tsx
index 668c37f663..1c7b5c177b 100644
--- a/packages/material-ui-lab/src/YearPicker/PickersYear.tsx
+++ b/packages/material-ui-lab/src/YearPicker/PickersYear.tsx
@@ -76,7 +76,7 @@ const PickersYearRoot = experimentalStyled(
       };
     },
   },
-)(({ styleProps }) => ({
+)<{ styleProps: { wrapperVariant: WrapperVariant }}>(({ styleProps }) => ({
   flexBasis: '33.3%',
   display: 'flex',
   alignItems: 'center',

```

That way we have correct typings per component + it would be required.

Reply:
Required internally sounds like the better solution right now.

Reply:
There is some case that styleProps is not needed in simple component. That case we can go with this, right?

```js
<StyledComponent
  styleProps={{}}
  {...other}
/>
```

Reply:
@siriwatknp would you be willing to open separate PR with this change so that we can realistically see what use-cases are there?

Reply:
Opened https://github.com/mui-org/material-ui/pull/26073 There are over 70 issues currently in the docs, as the utility is used there as well. How about we re-export the `experimentalStyled()` as `internalStyled()` too and change the typings on this one? Or we can still do https://github.com/mui-org/material-ui/pull/25949#discussion_r622892282 we would even have correct typings on the `styleProps`

Reply:
In conclusion, I'd say we should go with the following change at this moment:

```diff index 668c37f663..1c7b5c177b 100644
--- a/packages/material-ui-lab/src/YearPicker/PickersYear.tsx
+++ b/packages/material-ui-lab/src/YearPicker/PickersYear.tsx
@@ -76,7 +76,7 @@ const PickersYearRoot = experimentalStyled(
       };
     },
   },
-)(({ styleProps }) => ({
+)<{ styleProps: { wrapperVariant: WrapperVariant }}>(({ styleProps }) => ({
   flexBasis: '33.3%',
   display: 'flex',
   alignItems: 'center',
```

I saw already some issues for `experimentalStyled` and people starting using it outside of our repo, so I don't think we should go ahead and make the `styleProps` required ad this moment. We have the generic for altering it internally. We can introduce later one of the proposals in https://github.com/mui-org/material-ui/pull/26073 but I don't know if this is that critical for blocking all ongoing PRs for the migration.